examples/plugins/ebpf/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+func main() {
+	hostname, _ := os.Hostname()
+	var (
+		addr   = flag.String("addr", "/var/run/scope/plugins/ebpf.sock", "unix socket to listen for connections on")
+		hostID = flag.String("hostname", hostname, "hostname of the host running this plugin")
+	)
+	flag.Parse()
+
+	log.Println("Starting...")
+
+	// Compile and run the ebpf script
+	ebpf := exec.Command("http-parse-simple.py")
+	ebpf.Stderr = os.Stderr
+	stdout, err := ebpf.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := ebpf.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if ebpf.Process != nil {
+			ebpf.Process.Kill()
+		}
+	}()
+
+	os.Remove(*addr)
+	listener, err := net.Listen("unix", *addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		listener.Close()
+		os.Remove(*addr)
+	}()
+
+	log.Printf("Listening on: unix://%s", *addr)
+
+	plugin := &Plugin{HostID: *hostID}
+	go plugin.loop(stdout)
+	http.HandleFunc("/", plugin.Handshake)
+	http.HandleFunc("/report", plugin.Report)
```

Comment:
As we discussed offline, having the plugin expose a `/report` endpoint instead of doing it the other way around (the Scope probe exposing it) implies information needs to be polled. For instant updates and efficiently processing large amounts of data (i.e. large reports) you probably want a way for the plugin to convey an initial state and send deltas.

Considering the nature of reports and how we use them it seems fine to simply do polling but it's very inefficient for some cases. Here are two examples:

* TCP connection plugin (https://github.com/weaveworks/scope/issues/1168): you probably don't want to report all the connections all the time. It may be worth reporting the initial connections and then a stream of updates.
* kube-proxy status plugin (for eliding it https://github.com/weaveworks/scope/issues/538): you probably want to report the initial state of the proxy and them a stream of updates with the connections being created/destroyed.

I am thinking that leaking the report data structure (and implicitly the underlying semantics of its merge operations) in the plugins may not be a good idea. For one, it's not self contained (you need to understand what a report is and topologies are to fully understand it) plus it has the efficiency limitations I mention above. I don't have a good answer for how the plugin interface should be but we can work on it.